### DISCUSSION TITLE
Saving the data to Db

### DISCUSSION THREAD

**Discussion by muudi10** [react-page/react-page#1253] [Q&A]
Hi, I have come across this package and was impressed by its simplicity and features, 
I am developing course creation platform where we have an admin and instructors to create courses and this in turn will be served from database via API to be consumed on front end so that students can consume the courses. 
I could not make much sense on docs. 
the few questions that I have revolve around saving the data and then consuming that data. 
I am hoping to use either  mongodb or dynamo db. 
I assume the text editor outputs a json object and but I am not sure how this is exposed ?
How will then the json object be used by the front end ?
does anyone have a working example using react admin that is then connected to db.
Many thanks

**Comment by macrozone** [comment]
its best to think about ReactPage as a controlled `<input />` field:

```
export default function EditorExample() {
  const defaultValueFromDB = useValueFromDB()
  const [value, setValue] = useState<Value | null>(); 
  useEffect(() => setValue(defaultValueFromDB), [defaultValueFromDB]) // depends on how you load it from the api
  
  const save = () => {
      saveToDB(value) // could be POST request, graphql mutation, etc.
  }

  return (
    <PageLayout>
      <Editor cellPlugins={cellPlugins} value={value} onChange={setValue} />
      <Button onClick={save}>save</Button>
    </PageLayout>
  );
}
```

if you just want to display the value:

```
export default function ReadOnly() {
  const defaultValueFromDB = useValueFromDB()
  
  return (
    <PageLayout>
      <Editor cellPlugins={cellPlugins} value={defaultValueFromDB} readOnly />
    </PageLayout>
  );
}
```
